src/megaapi_impl.cpp

```diff
@@ -21936,6 +22058,25 @@ void MegaApiImpl::sendPendingRequests()
             }
 #endif
             AttrMap attrs;
+            Node *previousNode = client->childnodebyname(parentNode, utf8Name, true);
+            if (!client->versions_disabled && previousNode)
+            {
+               assert(previousNode->type == FILENODE);
+               nameid favnid = AttrMap::string2nameid("fav");
+               auto it = previousNode->attrs.map.find(favnid);
+               if (it != previousNode->attrs.map.end())
+               {
+                   attrs.map[favnid] = it->second;
+               }
+
+               nameid lblnid = AttrMap::string2nameid("lbl");
+               it = previousNode->attrs.map.find(lblnid);
+               if (it != previousNode->attrs.map.end())
+               {
+                   attrs.map[lblnid] = it->second;
+               }
+            }
```

Comment:
Isn't this block duplicated? If so, perhaps it's better to encapsulate the code in a method, and use it wherever it's needed.

Reply:
Done in commit: ec73f8b66879e3437fecdf3942815196b147d013